src/controller/command-handlers/certificatereader.cpp

```diff
@@ -93,11 +94,17 @@ getCertificateWithStatusAndInfo(CardInfo::ptr card, const CertificateType certif
 CertificateReader::CertificateReader(const CommandWithArguments& cmd) : CommandHandler(cmd)
 {
     validateAndStoreOrigin(cmd.second);
+    static const QStringList SUPPORTED_LANGS {"en", "et", "fi", "ru"};
+    const QString lang = cmd.second.value(QStringLiteral("lang")).toString();
+    if (SUPPORTED_LANGS.contains(lang)) {
+        Application* app = qobject_cast<Application*>(qApp);
+        app->loadTranslations(QLocale(lang));
+    }
```

Comment:
I would say that supported languages logic does not belong to `CertificateReader` by [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single-responsibility_principle).

Would a `void checkAndLoadTranslations(const QString& lang)` in `application.hpp` be a better place? Then this location would become a simple two-liner:

```suggestion
    const QString lang = cmd.second.value(QStringLiteral("lang")).toString();
    loadTranslationsForSupportedLanguage(lang);
```